src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -912,9 +910,11 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
-  public partitionsOf = async (params: PartitionsOfParams): Promise<string[]> => {
+  public partitionsOf = async (params: PartitionsOfParams): Promise<Partition[]> => {
     const partitions = await (await this.contract).partitionsOf.callAsync(params.tokenHolder);
-    return bytes32ArrayToStringArray(partitions);
+    return bytes32ArrayToStringArray(partitions).map(element => {
+      return element === Partition.Unlocked ? Partition.Unlocked : Partition.Undefined;
```

Comment:
This check should be refactored into some kind of utility function (to be easily extensible later when more partitions are added). The function can be called `parsePartitionValue` or something along those lines, and should

- Receive the string value
- Use a switch statement and check for each possible partition value, returning the corresponding value when it matches
- Special case for '0': return `Partition.Undefined`
- Default case: throw an error. This case means the core team added a new partition and our wrappers haven't been adapted